### ISSUE TITLE
function `ft_open_ex()` has a potential panic issue without pre-checking

### ISSUE THREAD

**Issue by cicilzx** [ftdi-rs/libftd2xx#83]
Hi, I'm developing a fuzzer to test the safe abstraction, and I found a panic bug when calling function `src/lib/ft_open_ex()`. Below is a test case:
```rust
#[allow(warnings, unused, non_snake_case)]
#[test]
fn test_ft_open_ex_muIpv() {
    let arg: String = String::from("\0\u{e}.r");
    let flag: u32 = 3590405307;
    let _ = ft_open_ex(arg.as_str(), flag);
}
```
Output:
```
thread 'test_ft_open_ex_muIpv' panicked at src/lib.rs:2008:48:
called `Result::unwrap()` on an `Err` value: NulError(0, [0, 14, 46, 114])
stack backtrace:
   0: rust_begin_unwind
             at /rustc/d84b9037541f45dc2c52a41d723265af211c0497/library/std/src/panicking.rs:652:5
   1: core::panicking::panic_fmt
             at /rustc/d84b9037541f45dc2c52a41d723265af211c0497/library/core/src/panicking.rs:72:14
   2: core::result::unwrap_failed
             at /rustc/d84b9037541f45dc2c52a41d723265af211c0497/library/core/src/result.rs:1658:5
   3: core::result::Result<T,E>::unwrap
             at /rustc/d84b9037541f45dc2c52a41d723265af211c0497/library/core/src/result.rs:1081:23
   4: libftd2xx::ft_open_ex
             at ./src/lib.rs:2008:20
   5: libftd2xx::test_ft_open_ex_muIpv
             at ./src/lib.rs:2504:13
   6: libftd2xx::test_ft_open_ex_muIpv::{{closure}}
             at ./src/lib.rs:2480:27
   7: core::ops::function::FnOnce::call_once
             at /rustc/d84b9037541f45dc2c52a41d723265af211c0497/library/core/src/ops/function.rs:250:5
   8: core::ops::function::FnOnce::call_once
             at /rustc/d84b9037541f45dc2c52a41d723265af211c0497/library/core/src/ops/function.rs:250:5
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
I think checking the scope of `arg` when calling the `ffi::CString` would make the external API more robust.
https://github.com/ftdi-rs/libftd2xx/blob/da7c8773110de2b9aaeac47325f2aeca07d7e88d/src/lib.rs#L2008

**Comment by newAM** [comment]
Good catch, thank you!

My laptop and desktop are all packed up since I'm moving, but I will get to this in a couple weeks!

**Comment by newAM** [comment]
Sorry this took a while to get to!

Fixed in release 0.33.1
